lib/chat_api_web/controllers/email_conversation_controller.ex

```diff
@@ -0,0 +1,91 @@
+defmodule ChatApiWeb.EmailConversationController do
+  use ChatApiWeb, :controller
+
+  alias ChatApi.{Conversations, Messages, Customers}
+  alias Customers.{Customer}
+
+  action_fallback(ChatApiWeb.FallbackController)
+
+  @spec create(Conn.t(), map()) :: {:error, Ecto.Changeset.t()} | Plug.Conn.t()
+  def create(
+        conn,
+        %{
+          "account_id" => account_id,
+          "body" => body,
+          "customer_id" => customer_id,
+          "email_address" => email_address
+        } = params
+      ) do
+    subject = Map.get(params, "subject")
+    customer = Customers.get_customer!(customer_id)
+
+    # TODO: handle failures
+    with {:ok, %Customer{} = _customer} <-
+           Customers.update_customer(customer, %{email: email_address}) do
+      create_conversation_and_send_message(conn, %{
+        "account_id" => account_id,
+        "body" => body,
+        "customer_id" => customer_id,
+        "email_address" => email_address,
+        "subject" => subject
+      })
+    end
+  end
+
+  def create(
```

Comment:
I'm still not quite used to the idea of pattern matching, but I'm using it here to pattern match on the existence of `customer_id` in the params. If it exists, then we assume that the customer exists in the database and we update it with the provided email address. If it doesn't, then we create the customer.